src/main/java/org/killbill/billing/catalog/api/StaticCatalog.java

```diff
@@ -70,6 +72,7 @@
      */
     public Plan createOrFindCurrentPlan(PlanSpecifier spec, PlanPhasePriceOverridesWithCallContext overrides) throws CatalogApiException;
 
+    // TODO_CATALOG All these exceptions are a result of having our DefaultVersionedCatalog implementation implement both StaticCatalog and Catalog (and are needed for the Catalog case)
```

Comment:
Maybe keep these comments outside of the API files (to avoid confusing the user), since they are implementation details.

Reply:
yep, and issue is gone anyways.